velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -627,6 +654,49 @@ void PageReader::makeDecoder() {
               type_->typeLength_,
               true);
           break;
+        case thrift::Type::INT64:
+          if (type_->logicalType_.has_value()) {
+            auto logicalType = type_->logicalType_.value();
+            if (logicalType.__isset.TIMESTAMP) {
+              VELOX_CHECK(
+                  logicalType.TIMESTAMP.isAdjustedToUTC,
+                  "Only UTC adjusted Timestamp is supported");
+              if (logicalType.TIMESTAMP.unit.__isset.MICROS) {
+                timestampDecoder_ = std::make_unique<
+                    dwio::common::TimestampDecoder>(
+                    dwio::common::TIMESTAMP_PRECISION::MICROS,
+                    std::make_unique<dwio::common::SeekableArrayInputStream>(
+                        pageData_, encodedDataSize_),
+                    false,
+                    parquetTypeBytes(type_->parquetType_.value()));
+              } else if (logicalType.TIMESTAMP.unit.__isset.MILLIS) {
+                timestampDecoder_ = std::make_unique<
+                    dwio::common::TimestampDecoder>(
+                    dwio::common::TIMESTAMP_PRECISION::MILLIS,
```

Comment:
It seems the difference between above two branches is about the timestamp precision. Maybe we can define a precision variable according to the unit in logicalType, and create the timestampDecoder. This could help remove duplicate code.